src/lib.rs

```diff
@@ -274,6 +275,40 @@ impl UltraLogLog {
     pub fn get_state(&self) -> &[u8] {
         &self.state
     }
+    /// Serializes UltraLogLog to a file using bincode
+    /// the serde feature must be enabled
+    pub fn save(&self, path: &str) -> std::io::Result<()> {
```

Comment:
I'd suggest returning a binary `Vec<u8>` instead of accepting a fs path. Leaving the IO operation to downstream users instead of handling them in this library. This can on the one hand make these methods more generic (maybe a system can buffer the content somewhere), and keep the scope concentrated, as "how to perform IO" is another big topic which beyond what this library should care about.